frontend/apps/reader/modules/readersearch.lua

```diff
@@ -177,22 +177,31 @@ function ReaderSearch:onShowFulltextSearchInput()
         end,
     }
 
-    if not self.ui.document.info.has_pages then
-        local checkbox_shift = math.floor((self.input_dialog.width - self.input_dialog._input_widget.width) / 2 + 0.5)
-        local check_buttons = HorizontalGroup:new{
+    local checkbox_shift = math.floor((self.input_dialog.width - self.input_dialog._input_widget.width) / 2 + 0.5)
+    local check_buttons
+    if self.ui.document.info.has_pages then
+        check_buttons = HorizontalGroup:new{
+            HorizontalSpan:new{width = checkbox_shift},
+            VerticalGroup:new{
+                align = "left",
+                self.check_button_case,
+            },
+        }
+    else
+        check_buttons = HorizontalGroup:new{
             HorizontalSpan:new{width = checkbox_shift},
             VerticalGroup:new{
                 align = "left",
                 self.check_button_case,
                 self.check_button_regex,
```

Comment:
You can avoid this duplication with (luckily, it is last so we can make it nil):
```
        check_buttons = HorizontalGroup:new{
            HorizontalSpan:new{width = checkbox_shift},
            VerticalGroup:new{
                align = "left",
                self.check_button_case,
                not self.ui.document.info.has_pages and self.check_button_regex or nil,
```
(not sure if the `or nil` is needed and if having it `false` is ok).

Reply:
`or nil` is necessary.